app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :check_user_for_edit_post, only: %i[edit]
+  before_action :check_user_for_delete_post, only: %i[destroy]
+
+  def index
+    @posts = Post.all
+    @post = Post.new
+    @main_user = User.find(current_user.id)
+  end
+
+  def show; end
+
+  def new
+    @post = Post.new
+  end
+
+  def edit
+    @post = Post.find(params[:id])
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      flash[:success] = 'Post has been successfully created'
+      redirect_back(fallback_location: root_path)
+    else
+      render 'posts/index'
+    end
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    if @post.update_attributes(post_params)
+      flash[:success] = 'post has been updated'
+      redirect_to root_path
+    else
+      render 'edit'
+    end
+  end
```

Comment:
Let's test for this commented out part of this method. 
Let's remember to always test for both negative and positive cases.

```ruby
  def update
    @post = Post.find(params[:id])
    if @post.update_attributes(post_params)
      flash[:success] = 'post has been updated'
      redirect_to root_path
    # else
    #   render 'edit'
    end
  end
```

Reply:
The same thing applies to this, I am taking off the else part of the if statement.